apps/dc_tools/odc/apps/dc_tools/_stac.py

```diff
@@ -171,19 +172,29 @@ def _get_stac_bands(
     grids = {}
     accessories = {}
 
+    self_link = _find_self_href(item)
+
     assets = item.get("assets", {})
 
-    def _get_path(asset):
-        path = asset["href"]
+    def _get_path(asset, force_relative=False):
+        path = URL(asset["href"])
         if relative:
-            path = Path(path).name
+            try:
+                path = path.relative_to(URL(self_link).parent)
+            except (ValueError, TypeError):
```

Comment:
`TypeError` looks suss? when does this happen?

Reply:
if it's when `self_link` is `None` then you should check that before attempting to compute parent of that link.

Reply:
Cleaned this up to make it more obviousl